### PR TITLE
core, dex, internal: block proposer syncing (first iteration)

### PR DESCRIPTION
**Pull request by Jsying**
Let a block proposer full node that is crash or stopped join consensus and start proposing block again.

### CODE REVIEW

File: dex/blockproposer.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	syncing   int32
+	proposing int32
+	dex       *Dexon
+	dMoment   time.Time
+
+	wg     sync.WaitGroup
+	stopCh chan struct{}
+}
+
+func NewBlockProposer(dex *Dexon, dMoment time.Time) *blockProposer {
```
Comment (aitjcize) [inline dex/blockproposer.go]: maybe we should store the `consensus` object in `BlockProposer` instead?

File: dex/blockproposer.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+}
+
+func (b *blockProposer) Start() error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	if !atomic.CompareAndSwapInt32(&b.running, 0, 1) {
+		return fmt.Errorf("block proposer is already running")
+	}
+	log.Info("Block proposer start")
```
Comment (aitjcize) [inline dex/blockproposer.go]: nit: `Block proposer started`

File: les/handler_test.go [hunk 3]
```diff
@@ -562,22 +562,24 @@ func TestTransactionStatusLes2(t *testing.T) {
...
-	// create a reorg that rolls them back
-	gchain, _ = core.GenerateChain(params.TestChainConfig, chain.GetBlockByNumber(0), ethash.NewFaker(), db, 2, func(i int, block *core.BlockGen) {})
-	if _, err := chain.InsertChain(gchain); err != nil {
-		panic(err)
-	}
-	// wait until TxPool processes the reorg
-	for i := 0; i < 10; i++ {
-		if pending, _ := txpool.Stats(); pending == 3 {
-			break
+	/*
```
Comment (aitjcize) [inline les/handler_test.go]: probably just remove these altogether? we don't need reorg in dexon.

**Comment by Jsying** [comment]
Rebased and fixed review comment

**Pull request merged**